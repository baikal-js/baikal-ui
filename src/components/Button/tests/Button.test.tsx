import { Button } from '../../index'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'

describe('tests for component Button', () => {
  const Component = ({ children = 'Click me!', ...args }) => <Button {...args}>{children}</Button>

  // props
  describe('should be rendered correctly', () => {
    test('variant bordered', () => {
      render(<Component variant='bordered' />)
      const button = screen.getByText('Click me!')
      expect(button.classList.contains('bui-btn-variant_bordered')).toBeTruthy()
    })

    test('size large', () => {
      const { container } = render(<Component size='large' />)
      const button = container.querySelector('button')
      expect(button.classList.contains('bui-btn-size_large')).toBeTruthy()
    })

    test('button disabled', () => {
      const { asFragment } = render(<Component disabled />)
      expect(asFragment()).toMatchSnapshot()
    })
  })

  // onClick & disabled
  describe('should be clickable', () => {
    test('on click button active', () => {
      const onClick = jest.fn()
      render(<Component onClick={onClick} />)
      fireEvent.click(screen.getByText('Click me!'))
      expect(onClick).toHaveBeenCalledTimes(1)
    })

    test('on click button disabled', () => {
      const onClick = jest.fn()
      render(<Component onClick={onClick} disabled />)
      fireEvent.click(screen.getByText('Click me!'))
      expect(onClick).not.toHaveBeenCalled()
    })
  })

  // snapshots
  describe('should be mounted', () => {
    const onClick = jest.fn()
    test('mounted active button', () => {
      const { asFragment } = render(
        <Component onClick={onClick} size='small' variant='bordered'>
          Active button
        </Component>,
      )
      expect(asFragment()).toMatchSnapshot()
    })

    test('mounted disabled button', () => {
      const { asFragment } = render(
        <Component onClick={onClick} size='middle' variant='filled' disabled>
          Disabled button
        </Component>,
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
