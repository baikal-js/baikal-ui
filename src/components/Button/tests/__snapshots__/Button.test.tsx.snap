// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for component Button should be mounted mounted active button 1`] = `
<DocumentFragment>
  <button
    class="bui-btn bui-btn-variant_bordered bui-btn-size_small "
    type="button"
  >
    Active button
  </button>
</DocumentFragment>
`;

exports[`tests for component Button should be mounted mounted disabled button 1`] = `
<DocumentFragment>
  <button
    class="bui-btn bui-btn-variant_filled bui-btn-size_middle bui-btn-disabled_filled bui-btn-disabled"
    disabled=""
    type="button"
  >
    Disabled button
  </button>
</DocumentFragment>
`;
